Framework/Crystal/src/SaveHKL.cpp

```diff
@@ -441,11 +441,23 @@ void SaveHKL::exec() {
         if (cosines) {
           out << std::setw(8) << std::fixed << std::setprecision(5) << lambda;
           out << std::setw(8) << std::fixed << std::setprecision(5) << tbar;
-          Kernel::DblMatrix oriented = p.getGoniometerMatrix();
+          Kernel::DblMatrix R = p.getGoniometerMatrix();
 
-          auto U = ol.getU();
-          auto RU = oriented * U;
-          RU.Transpose();
+          R.Transpose();
+
+          DblMatrix T = ol.getUB();
+
+          V3D t1 = T * V3D(1, 0, 0);
```

Comment:
Aren't `tn` just the nth column of T?  It doesn't look like there's a `getCol` method for a `DblMatrix` but one could always init the `V3D` by accessing the values directly?

Reply:
They are the columns of `T` and you are right, there is no `getCol` method. I initialize the V3D explicitly now.